components/public-api-server/pkg/identityprovider/cache.go

```diff
@@ -225,18 +250,23 @@ func (rc *RedisCache) Signer(ctx context.Context) (jose.Signer, error) {
 		err := rc.persistPublicKey(ctx, rc.current)
 		if err != nil {
 			log.WithField("keyID", rc.currentID).WithError(err).Error("cannot repersist public key")
-			return nil, err
+			return err
 		}
 	}
+	return nil
+}
 
-	return jose.NewSigner(jose.SigningKey{
-		Algorithm: jose.RS256,
-		Key:       rc.current,
-	}, &jose.SignerOptions{
-		ExtraHeaders: map[jose.HeaderKey]interface{}{
-			jose.HeaderKey("kid"): rc.currentID,
-		},
-	})
+func (rc *RedisCache) sync(ctx context.Context, period time.Duration) {
+	_ = rc.reconcile(ctx)
+	ticker := time.NewTicker(period)
+	for ctx.Err() == nil {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			_ = rc.reconcile(ctx)
+		}
+	}
```

Comment:
nit (slightly more idiomatic)

```suggestion
	for {
		select {
		case <-ctx.Done():
			return
		case <-ticker.C:
			_ = rc.reconcile(ctx)
		}
	}
```